File: restaurantservice/order-service/src/main/java/pl/restaurant/orderservice/api/request/GenerateChartOptions.java
package pl.restaurant.orderservice.api.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import pl.restaurant.orderservice.api.response.chart.ChartName;
import pl.restaurant.orderservice.api.response.chart.OrderType;
import pl.restaurant.orderservice.api.response.chart.PeriodType;

import javax.validation.constraints.NotEmpty;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class GenerateChartOptions {
    private PeriodType periodType;

    @NotEmpty(message = "Okres nie może być pusty")
    private String period;
    private Long placeId;
    private ChartName chartName;
    private OrderType orderType;
}

File: restaurant-app/src/app/model/chart/generate-chart-options.ts
import { OrderType } from "../order-type";
import { PeriodType } from "../period-type";
import { ChartName } from "./chart-name";

export interface GenerateChartOptions {
  periodType: PeriodType;
  period: any;
  placeId: any;
  chartName: ChartName;
  orderType: OrderType;
}